doc/system-administration.md

```diff
@@ -179,10 +179,91 @@ There should be an entry for `certbot renew`. That command may be run manually a
+
+In an environment without Security Groups, a firewall must be configured on each machine to allow such restrictions. The firewall that comes configured with DRIVER by default is called `ufw`. The following is a set of general troubleshooting steps that may be performed when the servers are rebooted and the application no longer seems to behaving properly.
+
+
+### 1. Ensure ufw is running
+
+The firewall needs to be running on each of the three machines. Log in to each of the servers and run the command:
+```
+sudo ufw status
+```
+This should return a Status of `active`. If that's not the case, it can be activated with:
+```
+sudo ufw enable
+```
+
+
+### 2. Ensure driver-app is running
+
+Run the following command on the App machine to see the status of `driver-app`:
+```
+service driver-app status
+```
+This should return a status of `start/running`. If it is instead `stop/waiting`, start it with the command:
+```
+service driver-app restart
+```
+Then check the status again to ensure it comes up. If there are any problems here, the log file (`/var/log/upstart/driver-app.log`) may be opened to do allow further debugging. One common source of errors here is related to Docker not being able to start up properly. If this is the case, there will be a log message such as:
+```
+docker: Error response from daemon: failed to create endpoint driver-app on network bridge: iptables failed: iptables --wait -t filter -A DOCKER ! -i docker0 -o docker0 -p tcp -d 172.17.0.2 --dport 4000 -j ACCEPT: iptables: No chain/target/match by that name.
+```
+Restarting Docker can be performed by running:
+```
+service docker restart
+```
+After Docker is running, `driver-app` will then need to be restarted again.
+
+
+### 3. Ensure Redis is running
+
+Redis needs to have open permissions due to its usage via Windshaft. It is therefore paramount that network traffic is locked down via `ufw` or Security Groups. If the machine is running without a firewall in place for an extended period of time, there is a chance an automated system will be able to write data to it and cause a corrupt state. Once it's verified that `ufw` is running, check for any peculiar logs in `/var/log/redis/redis-server.log`, and check if Redis is running with the command:
+```
+sudo service redis-server status
+```
+If Redis is down, or the logs are showing any kind of disk persistence errors, it may be restarted with the command:
+```
+sudo service redis-server restart
+```
+There may also be problems with restarting `driver-app` in the above step if Redis is not behaving properly. If the App machine is showing Redis-related errors in `/var/log/upstart/driver-app.log`, then be sure to restart Redis before attempting to restart `driver-app`.
```

Comment:
> [...] then be sure to restart Redis before attempting to restart `driver-app`.

Can you think of any way to check if we'd experience Redis related errors before bringing `driver-app` back up? It'd be good if we could insert something about detecting/fixing before that step if we can (or else have an alternate path here for people who already brought `driver-app` up that might be blocking a Redis restart), that way a person going through these step by step doesn't run into an unpleasant surprise here